src/qibo/tomography_RGD/projectors.py

```diff
@@ -0,0 +1,978 @@
+            try:
+                labels.remove(Iden)
+            except:
+                Go_Delete = 0
+
+        # ------------------------------------- #
+        #   to pad up labels up to num_label    #
+        # ------------------------------------- #
+        while len(labels) < size:
+            Add = [generate_random_label(n, symbols)]
+            if Add != Iden:
+                labels = labels + Add
+
+    # ----------------------------------------- #
+    #   to keep just enough num_label  labels   #
+    # ----------------------------------------- #
+    labels = labels[:size]
+    return labels
+
+
+## ------------------------------------------ ##
+##    constructing the matrix from labels     ##
+## ------------------------------------------ ##
+
+
+# Generate a projector by accumulating the Kronecker products of Pauli matrices
+# XXX Used basically to make sure that our fast implementation is correct
+def build_projector_naive(label, label_format="big_endian"):
+    """to directly generate a Pauli matrix from tensor products
+
+    Args:
+        label (str): label of the projection, e.g.  'XXZYZ'
+        label_format (str, optional): the ordering of the label. Defaults to 'big_endian'.
+
+    Raises:
+        Exception: when the matrix size is too big (i.e. for qubit number > 6)
+
+    Returns:
+        ndarray: a matrix representing the Pauli operator
+    """
+    if label_format == "little_endian":
+        label = label[::-1]
+    if len(label) > 6:
+        raise Exception("Too big matrix to generate!")
+    projector = reduce(
+        lambda acc, item: np.kron(acc, item),
+        [matrix_dict[letter] for letter in label],
+        [1],
+    )
+    return projector
+
+
+# Generate a projector by computing non-zero coordinates and their values in the matrix, aka the "fast" implementation
+def build_projector_fast(label, label_format="big_endian"):
+    """to fastly generate a Pauli projection matrix in sparse matrix format
+
+    Args:
+        label (str): label of the projection, e.g.  'XXZYZ'
+        label_format (str, optional): the ordering of the label. Defaults to 'big_endian'.
+
+    Returns:
+        sparse matrix: sparse matrix of the Pauli operator representing label
+    """
+    if label_format == "little_endian":
+        label = label[::-1]
+
+    n = len(label)
+    d = 2**n
+
+    # map's result NOT subscriptable in py3, just tried map() -> list(map()) for py2 to py3
+    ij = [
+        list(map(binarize, y))
+        for y in [zip(*x) for x in product(*[ij_dict[letter] for letter in label])]
+    ]
+    values = [
+        reduce(lambda z, w: z * w, y)
+        for y in [x for x in product(*[values_dict[letter] for letter in label])]
+    ]
+    ijv = list(map(lambda x: (x[0][0], x[0][1], x[1]), zip(ij, values)))
+
+    i_coords, j_coords, entries = zip(*ijv)
+
+    projector = sparse.coo_matrix(
+        (entries, (i_coords, j_coords)), shape=(d, d), dtype=complex
+    )
+    return projector

```

Comment:
@BrunoLiegiBastonLiegi this is a case where it would be convenient to have the `quantum_info.basis.pauli_basis` function return a generator because then one could use that to directly access a specific element of the basis.